packages/mui-material/src/TablePagination/TablePagination.js

```diff
@@ -199,7 +199,7 @@ const TablePagination = React.forwardRef(function TablePagination(inProps, ref)
         {rowsPerPageOptions.length > 1 && (
           <TablePaginationSelect
             variant="standard"
-            input={<InputBase />}
+            {...(!SelectProps.variant && { input: <InputBase /> })}
```

Comment:
`InputBase` should not always be rendered in `TablePaginationSelect`. If variant is passed through `SelectProps` prop, that particular variant input should get rendered instead.

Reply:
I wonder why the `<InputBase />` is needed in the first place 🤔

Reply:
It's needed. Otherwise, `<Input />` is rendered. `InputBase` is rendered when no Select variant is provided.

Reply:
Confirmed, the `InputBase` is needed.